rinex/src/navigation/ephemeris.rs

```diff
@@ -364,12 +364,18 @@ impl Ephemeris {
      * Either by solving Kepler equations, or directly if such data is available.
      */
     pub(crate) fn sv_position(&self, sv: &Sv, epoch: Epoch) -> Option<(f64, f64, f64)> {
-        if let Some(pos_x) = self.get_orbit_f64("satPosX") {
-            if let Some(pos_y) = self.get_orbit_f64("satPosY") {
-                if let Some(pos_z) = self.get_orbit_f64("satPosZ") {
-                    //TODO PZ90=>ECEF transform should be applied, in case of GLONASS
-                    //     need to check for GEO/SBAS
-                    return Some((pos_x, pos_y, pos_z));
+        if let Some(pos_x_km) = self.get_orbit_f64("satPosX") {
```

Comment:
This could be written as an separate function / lambda to use `?`, but it's fine like this too.

Reply:
The thing here, it should not "return" in case those fields are missing. When those fields are missing, we should execute `kepler2ecef` that comes right after this if/else

Reply:
Yeah, that's what I meant by moving the code to a new function:

```rust
self.sat_position().or_else(|| self.kepler2ecef(sv, epoch))
```